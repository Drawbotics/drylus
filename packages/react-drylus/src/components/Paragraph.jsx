import React from 'react';
import PropTypes from 'prop-types';
import sv from '@drawbotics/drylus-style-vars';
import { css } from 'emotion';


const styles = {
  root: css`
    color: ${sv.colorPrimary};
    line-height: calc(${sv.defaultLineHeight} * 1.5);
  `,
};


const Paragraph = ({ children }) => {
  return <p className={styles.root}>{children}</p>;
};


Paragraph.propTypes = {
  /** Text displayed by the paragraph */
  children: PropTypes.node,
};


export default Paragraph;
