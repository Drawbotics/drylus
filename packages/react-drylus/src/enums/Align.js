import Enum from '@drawbotics/enums';


const Align = new Enum(
  'LEFT',
  'CENTER',
  'RIGHT',
);


export default Align;
