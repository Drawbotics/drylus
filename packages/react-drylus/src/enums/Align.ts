export enum Align {
  LEFT = 'LEFT',
  CENTER = 'CENTER',
  RIGHT = 'RIGHT',
  JUSTIFY = 'JUSTIFY',
}
