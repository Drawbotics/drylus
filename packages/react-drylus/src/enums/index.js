export Category from './Category';
export Tier from './Tier';
export Size from './Size';
export Position from './Position';
export Align from './Align';
export Color from './Color';
export Shade from './Shade';
