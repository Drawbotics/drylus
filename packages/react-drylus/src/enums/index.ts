export * from './Category';
export * from './Tier';
export * from './Size';
export * from './Position';
export * from './Align';
export * from './Color';
export * from './Shade';
