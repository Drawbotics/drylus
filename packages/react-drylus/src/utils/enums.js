import camelCase from 'lodash/camelCase';

import { Category, Color, Shade, Tier } from '../enums';

export function getEnumAsClass(enumVal) {
  return camelCase(enumVal?.description?.toLowerCase());
}

export function colorEnumToCategory(enumVal) {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
  return enumVal;
}

export function categoryEnumToColor(enumVal) {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
  return enumVal;
}

export function shadeEnumToTier(enumVal) {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
  return enumVal;
}

export function getIconForCategory(category) {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
