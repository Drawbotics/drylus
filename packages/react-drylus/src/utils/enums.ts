import camelCase from 'lodash/camelCase';

import { Category, Color, Position, Shade, Size, Tier } from '../enums';

type Enum = Category | Color | Shade | Tier | Size | Position;

export function getEnumAsClass<T extends Record<string, string>>(enumVal?: Enum): keyof T {
  return camelCase(enumVal?.toLowerCase());
}

export function colorEnumToCategory(enumVal: Color): Category | undefined {
  switch (enumVal) {
    case Color.RED:
      return Category.DANGER;
    case Color.GREEN:
      return Category.SUCCESS;
    case Color.ORANGE:
      return Category.WARNING;
    case Color.BLUE:
      return Category.INFO;
  }
}

export function categoryEnumToColor(enumVal: Category): Color | undefined {
  switch (enumVal) {
    case Category.DANGER:
      return Color.RED;
    case Category.SUCCESS:
      return Color.GREEN;
    case Category.WARNING:
      return Color.ORANGE;
    case Category.INFO:
      return Color.BLUE;
  }
}

export function shadeEnumToTier(enumVal: Shade): Tier | undefined {
  switch (enumVal) {
    case Shade.DARK:
      return Tier.PRIMARY;
    case Shade.MEDIUM:
      return Tier.SECONDARY;
    case Shade.LIGHT:
      return Tier.TERTIARY;
  }
}

export function getIconForCategory(category: Category): string {
  switch (category) {
    case Category.DANGER:
      return 'alert-circle';
    case Category.SUCCESS:
      return 'check-circle';
    case Category.WARNING:
      return 'alert-triangle';
    default:
      return 'info';
  }
}
