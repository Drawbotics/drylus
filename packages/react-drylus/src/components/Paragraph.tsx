import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Align } from '../enums';
import { Responsive, Style } from '../types';
import { useResponsiveProps } from '../utils';

const styles = {
  root: css`
    color: ${sv.colorPrimary};
    line-height: calc(${sv.defaultLineHeight} * 1.5);
  `,
  alignRight: css`
    text-align: right;
  `,
  alignCenter: css`
    text-align: center;
  `,
};

export interface ParagraphProps {
  /** Text displayed by the paragraph */
  children: React.ReactNode;

  /**
   * @default Align.LEFT
   */
  align?: Align;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Paragraph = ({ responsive, ...rest }: ParagraphProps) => {
  const { children, style, align = Align.LEFT } = useResponsiveProps<ParagraphProps>(
    rest,
    responsive,
  );
  return (
    <p
      style={style}
      className={cx(styles.root, {
        [styles.alignCenter]: align === Align.CENTER,
        [styles.alignRight]: align === Align.RIGHT,
      })}>
      {children}
    </p>
  );
};
