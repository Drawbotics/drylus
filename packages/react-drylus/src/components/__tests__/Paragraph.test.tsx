import React from 'react';
import { create } from 'react-test-renderer';

import { Align } from '../../enums';
import { Paragraph } from '../Paragraph';

describe('Paragraph', () => {
  describe('matches snapshot when', () => {
    it('is aligned to the left', () => {
      const tree = create(<Paragraph align={Align.LEFT}>Some content</Paragraph>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is aligned to the right', () => {
      const tree = create(<Paragraph align={Align.RIGHT}>Some content</Paragraph>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('is center aligned', () => {
      const tree = create(<Paragraph align={Align.CENTER}>Some content</Paragraph>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
