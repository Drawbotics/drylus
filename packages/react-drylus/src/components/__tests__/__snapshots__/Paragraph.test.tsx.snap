// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph matches snapshot when is aligned to the left 1`] = `
.emotion-0 {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
}

<p
  className="emotion-0"
>
  Some content
</p>
`;

exports[`Paragraph matches snapshot when is aligned to the right 1`] = `
.emotion-0 {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
  text-align: right;
}

<p
  className="emotion-0"
>
  Some content
</p>
`;

exports[`Paragraph matches snapshot when is center aligned 1`] = `
.emotion-0 {
  color: #172b4e;
  line-height: calc(1.2 * 1.5);
  text-align: center;
}

<p
  className="emotion-0"
>
  Some content
</p>
`;
